src/nupic/engine/Link.hpp

```diff
@@ -394,70 +394,6 @@ namespace nupic
     using Serializable::read;
     void read(LinkProto::Reader& proto);
 
-
```

Comment:
None of these are implemented anywhere and their presence sans implementation causes symbol errors at runtime in Python.